vignettes/subtotals.Rmd

```diff
@@ -0,0 +1,149 @@
+---
+title: "Subtotals"
+description: "Modify subtotals and headings from R"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Subtotals}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+```{r load data, include=FALSE}
+## Because the vignette tasks require communicating with a remote host,
+## we do all the work ahead of time and save a workspace, which we load here.
+## We'll then reference saved objects in that as if we had just retrieved them
+## from the server
+library(crunch)
+load("vignettes.RData")
+options(width=120)
+```
+
+A common task in the Market research world is to collapse two or more categories together to see how the collapsed categories compare to one another. For example, if you asked people to rate their preference on a scale of 1-10 you might want to see how the people who provide a rating between 1 and 5 compare to those who rated it between 6 and 10. This goes by a number of names like _Top Box_, _Top Two Box_, or _Nets_. But in Crunch it is called _Subtotals_
+
+Subtotals can be applied to any Categorical Crunch Variables, and we can check if a variable has subtotals with the `subtotals()` function. If it returns `NULL`, there are no subtotals.
+
+```{r no subtotals, eval=FALSE}
+subtotals(ds$obamaapp)
+```
+```{r no subtotals print, echo = FALSE}
+sub_initial_subtotals
+```
+
+To add subtotals, we can save a list of `Subtotal` objects. Each `Subtotal` object has three things:
+
+* `name` the label to identify the subtotal
+* `categoires` the categories to add subtotal (here you can use either category names or category ids)
+* `after` the category that the subtotal should follow (again, either category names or category ids)
+
+```{r add some subtotals, eval = FALSE}
+subtotals(ds$DiversityImportant) <- list(
+    Subtotal(name = "Follows closely",
+             categories = c("Strongly closely", "Very closely"),
+             after = "Somewhat closely"),
+    Subtotal(name = "Generally disagree",
+             categories = c("Not very closely", "Not at all"),
+             after = "Not at all")
+)
+```
+
+Now, if we check `subtotals()`, we can see that we have saved them. In this output we see a few different aspects of subtotals: the `anchor` is the id of the category to put the subtotal after (matching the `after` argument in `Subtotal()`), name, aggregation functions and `args` which in the this case are the category ids to include in the subtotal.
+
+```{r new subtotals, eval = FALSE}
+subtotals(ds$manningknowledge)
+```
+```{r, echo = FALSE}
+sub_subtotals1
+```
+
+This shows up in the Categorical variable card like so: 
```

Comment:
This is pretty explicit already, especially with the photo, but it might be nice to toss webapp in here just to make clear what variable card we are referring to.